src/main/java/data/TaskManager.java

```diff
@@ -435,12 +432,20 @@ public void markManager(WeekView weekView, MonthView monthView, boolean inMonthV
      * @throws TaskManagerException If the date is not in the current month or week being viewed.
      */
     private static LocalDate findDateFromDayNumber(WeekView weekView, MonthView monthView,
-                                                   boolean inMonthView, int dayInt) throws TaskManagerException {
-        LocalDate date;
-        if (inMonthView) {
-            date = monthView.getStartOfMonth().plusDays(dayInt);
-            checkIfDateInCurrentMonth(date);
+            boolean inMonthView, int dayInt) throws TaskManagerException {
+      
+        if (dayInt < 1 || dayInt > 31) {
+            throw new TaskManagerException("Invalid day number. Day must be between 1 and 31.");
+        }
```

Comment:
Good catch on invalid dates given via input!

Reply:
Could we check that this resolves the issue for months  with less than 31 days?